src/Supervisor.cpp

```diff
@@ -1221,4 +1221,60 @@ ZunResult Supervisor::FadeOutMusic(f32 fadeOutSeconds)
     return ZUN_SUCCESS;
 }
 #pragma optimize("", on)
+
+#pragma optimize("", on)
+// this is for rebinding keys
+u8 *th06::Controller::GetControllerState()
+{
+    static u32 controllerData[32];
+    memset(&controllerData, 0, sizeof(controllerData));
+    if (g_Supervisor.controller == NULL)
+    {
+        // TODO: not tested
+        JOYINFOEX joyinfo;
+        memset(&joyinfo, 0, sizeof(JOYINFOEX));
+        joyinfo.dwSize = sizeof(JOYINFOEX);
+        joyinfo.dwFlags = JOY_RETURNALL;
+        if (joyGetPosEx(0, &joyinfo) == JOYERR_NOERROR)
+        {
+            for (u32 local_3c = joyinfo.dwButtons, local_40 = 0; local_40 < 32; local_40 += 1, local_3c >>= 1)
+            {
+                if ((local_3c & 1) != 0)
+                {
+                    *(u8 *)((int)controllerData + local_40) = 0x80;
+                }
+            }
+        }
+    }
+    else
+    {
+        HRESULT HVar2 = g_Supervisor.controller->Poll();
+        if (FAILED(HVar2))
+        {
+            int local_retryCount = 0;
```

Comment:
This should be typed `i32`.

Also, please don't use underscores in names. Name this `retryCount`.